### PR TITLE
(SIMP-3115) Enforce full module blacklisting

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Updated the simp::kmod_blacklist class to also fully disable the
  module loading
* Provided the ability to override the disabled modules locally on the
  system by default, but also allow for absolute enforcement
* Provided the ability to lock module loading if the underlying OS has
  the capability (requires associated simplib update)

SIMP-3115 #close

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
I just realized that making all of those kmod resources is silly. I'm going to just collapse it all into a single file statement and eliminate the batch of resources in the catalog.

**Comment by trevor-vaughan** [comment]
This is now ready for final review.

File: .fixtures.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ---
 fixtures:
   repositories:
+    # Kluge for testing the automatic cert generation
```
Comment (jeefberkey) [inline .fixtures.yml]: is this needed for this PR?

**Comment by nick-markowski** [comment]
Patching now

File: data/common.yaml [hunk 2]
```diff
@@ -131,6 +131,7 @@ simp::server::data:
   - pam::wheel
   - selinux
   - svckill
+  - fips
```
Comment (jeefberkey) [inline data/common.yaml]: This looks like a separate issue

Comment (trevor-vaughan) [inline data/common.yaml]: @jeefberkey It was just a corrected typo. Do you want me to remove it?

**Comment by trevor-vaughan** [comment]
@nick-markowski This should be good to go. Travis is failing because we need the updated `augeasproviders-sysctl` to go through.

**Comment by nick-markowski** [comment]
@trevor-vaughan I'll review what I can for augeas sysctl and re-test against it

File: data/common.yaml (hunk 2 above)
Comment (jeefberkey) [inline data/common.yaml]: This is definitely related to https://github.com/simp/pupmod-simp-fips/pull/5, and not a typo.

Comment (jeefberkey) [inline data/common.yaml]: But no

**Pull request merged**